ethash/src/lib.rs

```diff
@@ -136,6 +138,32 @@ impl EthashManager {
 	}
 }
 
+/// Convert an Ethash boundary to its original difficulty. Basically just `f(x) = 2^256 / x`.
+pub fn boundary_to_difficulty(boundary: &ethereum_types::H256) -> U256 {
+	assert!(!boundary.is_zero());
+
+	let d = U512::from(&**boundary);
```

Comment:
This method could be rewritten to use `difficulty_to_boundary` but we would have to do an extra conversion. Still, both definitions are next to another and are tested in relation to one another, so I think it's fine to have this duplicated.

Reply:
I think something like `fn(n: impl Into<U512>) -> U256` could work